Beta docs (#931)
microsoftgraph/entra-powershell: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaUserOAuth2PermissionGrant.md

# Get-EntraBetaUserOAuth2PermissionGrant

## Synopsis

Gets an oAuth2PermissionGrant object.

## Syntax

```powershell
Get-EntraBetaUserOAuth2PermissionGrant 
 -ObjectId <String> 
 [-All] 
 [-Top <Int32>] 
 [<CommonParameters>]
```

## Description

> The `Get-EntraBetaUserOAuth2PermissionGrant` cmdlet gets an oAuth2PermissionGrant object for the specified user in Microsoft Entra ID. Specify `ObjectId` parameter to retrieve an oAuth2PermissionGrant object.

Reviewer: ```suggestion
The `Get-EntraBetaUserOAuth2PermissionGrant` cmdlet gets an oAuth2PermissionGrant object for the specified user in Microsoft Entra ID. Specify `ObjectId` parameter to retrieve an oAuth2PermissionGrant object.

In delegated scenarios with work or school accounts, the signed-in user must be assigned a supported Microsoft Entra role or a custom role with a supported role permission. The following least privileged roles are supported for this operation.

- Application Administrator
- Application Developer
- Cloud Application Administrator
- Directory Writers
- Privileged Role Administrator
- User Administrator
- Directory Readers
- Global Reader
- Guest Inviter
```

Author: Changes done.

## Examples

### Example 1: Retrieve the OAuth2 permission grants for a user

```powershell
Connect-Entra -Scopes 'Directory.Read.All'
$UserId = (Get-EntraBetaUser -Top 1).ObjectId
Get-EntraBetaUserOAuth2PermissionGrant -ObjectId $UserId
```

```Output
Id                                                               ClientId                             ConsentType ExpiryTime
--                                                               --------                             ----------- ----------
HXFXwKLgoUC4rwbZbCDIdffW8XpadQNIoHik9aQxrVHR6StBYBRhQI7tzKID_LIV 00001111-aaaa-2222-bbbb-3333cccc4444 Principal   08-01-2024 10:0...
9uBzRwC0s0CFCDQN6O4Ik_fW8XpadQNIoHik9aQxrVHR6StBYBRhQI7tzKID_LIV 11112222-bbbb-3333-cccc-4444dddd5555 Principal   13-01-2024 08:0...
```

> This example retrieves the OAuth2 permission grants for a user using object ID parameter.
> 
> The first command gets the ID of a Microsoft Entra ID user by using the Get-EntraBetaUser (./Get-EntraBetaUser.md) cmdlet.
> The command stores the value in the $UserId variable.
> 
> The second command gets the OAuth2 permission grants for the user identified by $UserId.

Reviewer: ```suggestion
This example retrieves the OAuth2 permission grants for a user using the ObjectId parameter. Use the `Get-EntraBetaUser` cmdlet to obtain the `ObjectId` value.
```

Author: Changes done.
